test/metabase/test/util.clj

```diff
@@ -100,71 +94,92 @@
 
 (defn- rasta-id [] (user-id :rasta))
 
-(def ^:private with-temp-defaults-fns
-  {Card
-   (fn [_] {:creator_id             (rasta-id)
-            :database_id            (data/id)
-            :dataset_query          {}
-            :display                :table
-            :name                   (tu.random/random-name)
-            :visualization_settings {}})
+(defn- default-updated-at-timestamped
+  [x]
+  (merge {:updated_at (t/zoned-date-time)} x))
+
+(defn- default-created-at-timestamped
+  [x]
+  (merge {:created_at (t/zoned-date-time)} x))
```

Comment:
Wouldn't a conditional `assoc` be more REPL friendly e.g., when the map is (still) an array-map?

Reply:
hmmm, I never realized the difference between array-map vs hash-map until today!

I always thought {} creates a hashmap.


Anyway, in which case `x` is a hashmap? I'm gonna change it to assoc anyway but still want to know

Reply:
Until about 10 elements the maps are array-maps. From that point on associng a new entry would result in a hash-map. This is of course implementation detail that can change in future versions.